src/libraries/Common/src/System/Threading/Tasks/TaskTimeoutExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+namespace System.Threading.Tasks
+{
+    /// <summary>
+    /// Task timeout helper based on https://devblogs.microsoft.com/pfxteam/crafting-a-task-timeoutafter-method/
+    /// </summary>
+    internal static class TaskTimeoutExtensions
```

Comment:
@stephentoub I chose to copy the `TaskTimeoutExtensions.cs` instead of moving the file from the `Common/tests` folder, since I really only need this one method.

Would probably make sense to add a few unit tests for that?